libs/s25main/FindWhConditions.h

```diff
@@ -35,10 +35,10 @@ struct HasWareAndFigure : protected HasMinWares, protected HasFigure
     bool operator()(const nobBaseWarehouse& wh) const;
 };
 
-struct HasMinSoldiers
+struct HasSoldiers
```

Comment:
Maybe a better name: `HasAnyMatchingSoldier` as passing an array of ranks had me thinking this would return true if ALL are available, not ANY.